### ISSUE TITLE
Generate a rust struct for a GraphQL type?

### ISSUE THREAD

**Issue by antholeole** [graphql-rust/graphql-client#443]
I have a type like so:

```graphql
type AuthenticationOutput {
  access_token: String!
  admin: Boolean!
  refresh_token: String!
}
```

and I'd like to generate a Rust struct for it. The use case is that Hasura allows you to define web hooks, and defines the inputs / outputs of the web hooks as GraphQL types. Is there a way to generate the struct for this type? or does the type have to be attached to a query? Thanks!

**Comment by banool** [comment]
Did you have any luck with this? I'm just now going to look into the graphql_client_codegen crate and see what I can do.

**Comment by antholeole** [comment]
I did not, sorry. Best of luck.

**Comment by banool** [comment]
For anyone else who comes along, I got pretty far by forking this repo instead: https://github.com/banool/async-graphql-reverse/.

**Comment by tomhoule** [comment]
I agree that should probably a separate crate. Closing this as "not planned" because not in scope, but feel free to disagree and open a new issue.
